benchmark/internal_server.go

```diff
@@ -84,16 +85,22 @@ func (s *internalServer) HostPort() string {
 
 // Advertise advertises with Hyperbahn.
 func (s *internalServer) Advertise(hyperbahnHosts []string) error {
+	var err error
 	config := hyperbahn.Configuration{InitialNodes: hyperbahnHosts}
-	hc, err := hyperbahn.NewClient(s.ch, config, nil)
+	s.hc, err = hyperbahn.NewClient(s.ch, config, nil)
 	if err != nil {
 		panic("failed to setup Hyperbahn client: " + err.Error())
 	}
-	return hc.Advertise()
+	return s.hc.Advertise()
 }
 
 func (s *internalServer) Close() {
-	s.ch.Close()
+	if s.ch != nil {
```

Comment:
this is never `nil` right?

Reply:
yeah.. was trying to be defensive, but agreed that it's not necessary

Reply:
> Maybe we should add a simple test that uses this and the `WithTestServer` and verifies that there's no goroutine leaks after Close?

yea, but it seems like goleak isn't used in this code base. perhaps a follow up to add goleak would be useful.

Reply:
If we use `WithTestServer` and add a test in the tchannel package, it does goroutine leak checks already.